localstack/aws/protocol/parser.py

```diff
@@ -1063,17 +1062,21 @@ def _parse_shape(
         Special handling of parsing the shape for s3 object-names (=key):
         trailing '/' are valid and need to be preserved, however, the url-matcher removes it from the key
         we check the request.url to verify the name.
-        We decode the key for the comparison with `base_url` in case of special characters.
+        Trailing slashes are stripped from the path, so we need special logic to compare the parsed Key parameter
+        against the path and add the missing slashes
```

Comment:
nit: This comment is a bit confusing now, because it describes the removal of the trailing slashes twice.

Reply:
Oops, yes, will rewrite everything!